src/tsickle.ts

```diff
@@ -335,10 +335,13 @@ function addClutzAliases(
       // Use the localName for the export then, but publish under the external name.
       localName = declaration.propertyName.text;
     }
-    globalSymbols +=
-        `\t\texport {${localName} as module$contents$${clutzModuleName}_${symbol.name}}\n`;
-    nestedSymbols +=
-        `\t\texport {module$contents$${clutzModuleName}_${symbol.name} as ${symbol.name}}\n`;
+    const mangledName = `module$contents$${clutzModuleName}_${symbol.name}`;
+    globalSymbols += `\t\texport {${localName} as ${mangledName}}\n`;
+    // TODO(mprobst): in TS3.7, this could be just "export {localName};". However in TS3.5,
+    // localName resolves within the module, so exporting {localName} causes a circular definition
+    // error. The workaround is to import the mangled name.
```

Comment:
I reread this a bunch of time and still can't follow this comment.

Normally 'mangled' names means you did some damage to a name to avoid a conflict (like $clutz below), but here I think the mangledName var is just the Closure post-module-processing global name of the symbol, isn't it?  Or I think maybe this comment is saying that the $clutz mangling can be removed in TS37, and the mention of mangling here is unrelated to the variable named 'mangled' just above it... ?  Or is the comment referring to the code above it?

Reply:
`mangledName` refers to the name as it will be mangled by Closure. I think it's correct to describe Closure's post processing of the name into a single variable as mangling?

The comment says `in TS3.7, this could be just "export {localName};"` in reference to the lines below it (like comments usually do, right?). The intended meaning is that once tsickle is on TS3.7, you can replace the two lines below with `nestedSymbols += '\t\texport {${localName}}';`. I've updated the comment to say `Once tsickle is on TS3.7, the two lines below can be replaced with "export {localName};".`, hopefully that's explicit enough?

Reply:
I see.  I think I was confused by "The workaround is to import the mangled name.", where the workaround is to mangle the name with $clutz and use that mangled name to refer to the Closure-mangled name.  Two mangled names.